### PR TITLE
[#425] made code compileable with java 11

### PR DESCRIPTION
**Pull request by cdietrich**
[#425] made code compileable with java 11
Signed-off-by: Christian Dietrich <christian.dietrich@itemis.de>

### CODE REVIEW

File: org.eclipse.xtext.xbase.testing/src/org/eclipse/xtext/xbase/testing/OnTheFlyJavaCompiler2.xtend [hunk 1]
```diff
@@ -114,7 +114,7 @@ class OnTheFlyJavaCompiler2 {
 				}
 			}
 		''')
-		return clazz.newInstance as Functions.Function0<RT>
+		return clazz.constructor.newInstance as Functions.Function0<RT>
```
Comment (szarnekow) [inline org.eclipse.xtext.xbase.testing/src/org/eclipse/xtext/xbase/testing/OnTheFlyJavaCompiler2.xtend]: why not declaredConstructor?

File: org.eclipse.xtext.common.types/src/org/eclipse/xtext/common/types/util/AnnotationLookup.java [hunk 2]
```diff
@@ -52,6 +52,19 @@ public JvmAnnotationReference findAnnotation(/* @NonNull */ JvmAnnotationTarget
 		return null;
 	}
 	
+	public JvmAnnotationReference findAnnotation(/* @NonNull */ JvmAnnotationTarget annotationTarget, /* @NonNull */ String lookupType) {
```
Comment (szarnekow) [inline org.eclipse.xtext.common.types/src/org/eclipse/xtext/common/types/util/AnnotationLookup.java]: Should the method above delegate to this implementation here?

File: org.eclipse.xtext.xbase.testing/src/org/eclipse/xtext/xbase/testing/OnTheFlyJavaCompiler2.xtend (hunk 1 above)
Comment (cdietrich) [reply org.eclipse.xtext.xbase.testing/src/org/eclipse/xtext/xbase/testing/OnTheFlyJavaCompiler2.xtend]: cause i could not force push ;)
should be now declared one

File: org.eclipse.xtext.common.types/src/org/eclipse/xtext/common/types/util/AnnotationLookup.java (hunk 2 above)
Comment (cdietrich) [reply org.eclipse.xtext.common.types/src/org/eclipse/xtext/common/types/util/AnnotationLookup.java]: will change

**Comment by cdietrich** [comment]
@szarnekow ok now?

**Pull request merged**